cmd/prometheus/main.go

```diff
@@ -187,6 +187,9 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "extra-scrape-metrics":
 				c.scrape.ExtraMetrics = true
 				level.Info(logger).Log("msg", "Experimental additional scrape metrics enabled")
+			case "metadata-wal-records":
+				c.scrape.EnableMetadataStorage = true
+				level.Info(logger).Log("msg", "Experimental metadata records in WAL enabled, required for remote write 2.0")
```

Comment:
It would be fully enforced in a follow up and/or potentially rw2.0 not launching releasing until 3.0 which I hope we can have get metadata in WAL turned on by default, but as a stop gap we can have an intermediate follow up that checks the RW format and enablement of metadata in WAL, and fatal exists if there's no metadata in the WAL